pac4j-core/src/main/java/org/pac4j/core/credentials/extractor/FormExtractor.java

```diff
@@ -21,14 +23,14 @@ public FormExtractor(final String usernameParameter, final String passwordParame
     }
 
     @Override
-    public UsernamePasswordCredentials extract(WebContext context) {
+    public Optional<UsernamePasswordCredentials> extract(WebContext context) {
         final String username = context.getRequestParameter(this.usernameParameter);
```

Comment:
I'm surprised that getRequestParameter does not return an Optional :)

Reply:
As the return type is String, we could use empty one to denote absence. What you think?

Reply:
That it is a very bad idea ^^
I think that a request parameter can be present and have no value furthermore, but anyway, `Optional` is a good idea!